private/buf/buffetch/buffetch.go

```diff
@@ -93,6 +93,8 @@ type PathResolver interface {
 // Ref is an image file or source bucket reference.
 type Ref interface {
 	PathResolver
+	// ID is the Ref's ID.
```

Comment:
What does this mean? What is an ID? What is its form? What can I depend on and not depend on in terms of its form (likely nothing, likely that it's just an opaque string). ie explain all of this in the Godoc.

Since this is an internal package, an explanation of the form of each ID for each different `Ref` type, just for users' reference, would also be helpful here, but make it clear in the Godoc that the form of the ID is not stable and should not be relied on.